lib/irb/completion.rb

```diff
@@ -166,10 +166,10 @@ def self.retrieve_files_to_require_relative_from_current_dir
 
     def self.retrieve_completion_data(input, bind: IRB.conf[:MAIN_CONTEXT].workspace.binding, doc_namespace: false)
       case input
-      when /^((["'`]).*\2)\.([^.]*)$/
+      when /^(.*["'`])\.([^.]*)$/
```

Comment:
```suggestion
      # this regexp only matches the closing character because of irb's Reline.completer_quote_characters setting
      # details are described in: https://github.com/ruby/irb/pull/523
      when /^(.*["'`])\.([^.]*)$/
```